### PR TITLE
Enable players to choose policies

### PR DESCRIPTION
**Pull request by Dammic**
In this PR, choose policy modal has been added and with it, another game phase.

### CODE REVIEW

File: events/SocketEvents.js [hunk 1]
```diff
@@ -153,8 +156,76 @@ module.exports = function (io, RoomsManager) {
...
+            if (includes(drawnCards, choice)) {
+                const president = RoomsManager.getPresident(socket.currentRoom)
+                const chancellor = RoomsManager.getChancellor(socket.currentRoom)
+                if (gamePhase === GamePhases.PresidentPolicyChoice && president.playerName === socket.currentPlayerName) {
+                    RoomsManager.setGamePhase(socket.currentRoom, GamePhases.ChancellorPolicyChoice)
+                    const chancellorEmit = RoomsManager.getRoleSocket(socket.currentRoom, PlayerRole.ROLE_CHANCELLOR)
+                    io.sockets.in(socket.currentRoom).emit(SocketEvents.ChancellorChoosePolicy, {
+                        data: {
+                            timestamp: getCurrentTimestamp(),
+                            chancellorName: RoomsManager.getChancellor(socket.currentRoom).playerName,
```
Comment (Terverok) [inline events/SocketEvents.js]: You don't have to get actual Chancellor from RoomManager, you already did that earlier and assigned the player to **variable `chancellor`**

File: events/SocketEvents.js [hunk 2]
```diff
@@ -153,8 +156,76 @@ module.exports = function (io, RoomsManager) {
...
+                        },
+                    })
+                    RoomsManager.discardPolicy(socket.currentRoom, choice)
+                    drawnCards = RoomsManager.getDrawnCards(socket.currentRoom)
```
Comment (Terverok) [inline events/SocketEvents.js]: `drawnCards` isn't maybe in that case a reference to an arrray? That would mean you don't have to get it a second time.

File: events/SocketEvents.js [hunk 3]
```diff
@@ -153,8 +156,76 @@ module.exports = function (io, RoomsManager) {
...
+                    chancellorEmit(SocketEvents.ChoosePolicy, {
+                        data: {
+                            policyCards: drawnCards,
+                            title: 'Choose policy to enact',
+                        },
+                    })
+                } else if (gamePhase === GamePhases.ChancellorPolicyChoice && chancellor.playerName === socket.currentPlayerName) {
+                    const discardedCard = get(pullAt(drawnCards, indexOf(drawnCards, choice)), '0')
+                    RoomsManager.discardPolicy(socket.currentRoom, discardedCard)
```
Comment (Terverok) [inline events/SocketEvents.js]: Higher, when president has chosen card to discard you pass `choice` straight to `discardPolicy`, here you do this diffrently. _Please decide on what way is correct for you_ ;)
I know what you do here, just think that's inconsistent.

File: utils/RoomsManager.js [hunk 4]
```diff
@@ -295,6 +317,42 @@ class RoomsManager {
...
+            this.rooms_props[roomName].liberalPoliciesOnTheTable += 1
+        } else {
+            this.rooms_props[roomName].fascistPoliciesOnTheTable += 1
+        }
+        this.rooms_props[roomName].drawnCards = []
+    }
+    discardPolicy(roomName, card) {
+        const { drawnCards } = this.rooms_props[roomName]
+        pullAt(drawnCards, indexOf(drawnCards, card))
+        this.rooms_props[roomName].drawnCards = drawnCards
```
Comment (Terverok) [inline utils/RoomsManager.js]: `drawnCards` is a reference to an array? right?

File: utils/RoomsManager.js [hunk 5]
```diff
@@ -295,6 +317,42 @@ class RoomsManager {
...
+        return this.rooms_props[roomName].drawnCards
+    }
+
+    getChoicePolicyCards(roomName) {
+        const { drawPile, discardPile } = this.rooms_props[roomName]
+
+        let tmpDrawPile = drawPile
+        if (size(drawPile) < 3) {
+            tmpDrawPile = shuffle(concat(drawPile, discardPile))
+            this.rooms_props[roomName].drawPile = tmpDrawPile
```
Comment (Terverok) [inline utils/RoomsManager.js]: This assigment in unnecessary when in next step you execute `drop` and assign a new array.

File: utils/RoomsManager.js (hunk 4 above)
Comment (Dammic) [reply utils/RoomsManager.js]: Yes. Cards suggests that.

File: utils/RoomsManager.js (hunk 5 above)
Comment (Dammic) [reply utils/RoomsManager.js]: Ok, you are all left! Thanks

File: events/SocketEvents.js (hunk 1 above)
Comment (Dammic) [reply events/SocketEvents.js]: Good point - thanks!

File: events/SocketEvents.js [hunk 6]
```diff
@@ -153,8 +156,76 @@ module.exports = function (io, RoomsManager) {
...
+            io.sockets.in(socket.currentRoom).emit(SocketEvents.PresidentChoosePolicy, {
+                data: {
+                    timestamp: getCurrentTimestamp(),
+                    presidentName: RoomsManager.getPresident(socket.currentRoom).playerName,
+                },
+            })
+            presidentEmit(SocketEvents.ChoosePolicy, {
+                data: {
+                    policyCards: RoomsManager.getChoicePolicyCards(socket.currentRoom),
+                    title: 'Discard one policy and pass the to the chancellor',
```
[the remark this answers is not in the document]
Comment (Dammic) [reply events/SocketEvents.js]: Thanks! It's appreciated!

File: events/SocketEvents.js (hunk 2 above)
Comment (Dammic) [reply events/SocketEvents.js]: Extremelly good point! Thank you for that observation!

File: events/SocketEvents.js (hunk 3 above)
Comment (Dammic) [reply events/SocketEvents.js]: You are right - I've made front send the discarded card right now and we can have similar logic on backend. Thanks!

**Pull request merged**